site/zenodo_rdm/moderation/rules.py

```diff
@@ -65,11 +64,19 @@ def links_rule(identity, draft=None, record=None):
     extracted_links = extract_links(str(record.metadata))
 
     for link in extracted_links:
-        status = current_domain_tree.get_status(link)
-        if status == "banned":
-            score += current_scores.spam_link
-        elif status == "safe":
-            score += current_scores.ham_link
+        domain = LinkDomain.lookup_domain(link)
+        if domain is None:
+            continue
+        if domain.status == LinkDomainStatus.BANNED:
+            if domain.score is not None:
+                score += domain.score
+            else:
+                score += current_scores.spam_link
+        elif domain == LinkDomainStatus.SAFE:
+            if domain.score is not None:
+                score += domain.score
+            else:
+                score += current_scores.ham_link
```

Comment:
How do we score if the status is MODERATED?

Reply:
In practice, they contribute `0`, since it means that we have to have a look at them (i.e. "moderate" them). 

So tbh, I added the `MODERATED` status just to have a "other" value that can be used as a placeholder for ingesting new domains that we don't want to mark as banned/safe immediately.